examples/subscriptions/examples/subscribe_blocks.rs

```diff
@@ -0,0 +1,39 @@
+//! Example of subscribing to blocks and watching blocks.
+
+use alloy_network::Ethereum;
+use alloy_node_bindings::{Anvil, AnvilInstance};
+use alloy_provider::{Provider, RootProvider};
+use alloy_pubsub::PubSubFrontend;
+use alloy_rpc_client::RpcClient;
+use eyre::Result;
+use futures_util::{stream, StreamExt};
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let (provider, _anvil) = init().await;
+
+    let sub = provider.subscribe_blocks().await?;
+    let mut stream = sub.into_stream().take(2);
```

Comment:
Do we always call into_stream()? Should we return that in the subscribe_blocks return type instead @DaniPopes by `impl Stream` on the type?